src/lib/network_pool/snark_pool.ml

```diff
@@ -365,10 +370,14 @@ let%test_module "random set test" =
       let frontier_broadcast_pipe_r, _ =
         Broadcast_pipe.create (Some (Mocks.Transition_frontier.create ()))
       in
+      let verifier =
+        Async.Thread_safe.block_on_async_exn (fun () ->
```

Comment:
I kind of don't like this `block_on_async_exn`. Can you turn `gen` such that the type of it is `t Deferred.t Quickcheck.Generator.t`? Then, when you want to use the generator, you can use `Async.Quickcheck.test` rather than `Quickcheck.test`. `Async.Quickcheck.test` let's you have deferreds in your test.

Reply:
This is a good idea!